openbb_platform/providers/oecd/openbb_oecd/models/gdp_forecast.py

```diff
@@ -94,11 +96,18 @@ def extract_data(
         )
         data_df = data_df[data_df["country"] == query.country]
         data_df = data_df[["country", "date", "value"]]
+        data_df["date"] = data_df["date"].apply(helpers.oecd_date_to_python_date)
+        data_df = data_df[
+            (data_df["date"] <= query.end_date) & (data_df["date"] >= query.start_date)
+        ]
+        data_df["date"] = data_df["date"].apply(
+            lambda x: x.year
+        )  # Validator won't accept datetime.date?
```

Comment:
I think this is because the validator is expecting a string representation. Although, because you have already conformed the date field in the extract process, the validator is now redundant.

Reply:
I thought that too but it accepts str|date so it was unclear what was actually going wrong